src/core/terms.ml

```diff
@@ -59,6 +59,8 @@ type term =
   (** Wildcard (only used for surface matching, never in a LHS). *)
   | TRef of term option ref
   (** Reference cell (only used for surface matching). *)
+  | LLet of term * term * (term, term) Bindlib.binder
+  (** [LLet(t, a, u)] is [let x : a ≔ t in u] (with [x] bound in [u]). *)
```

Comment:
As mentioned in a previous comment, I think that it would be better to put the type first and write `LLet(a,t,u)`, like we do in abstractions and products.

Reply:
It's done, hoping I haven't forgotten any place.

Reply:
And by the way, in `iter` functions, the type is iterated on before the terms, as for products and abstractions.